modules/core/src/v2/coins/trx.ts

```diff
@@ -249,6 +275,200 @@ export class Trx extends BaseCoin {
     return Buffer.from(sig, 'hex');
   }
 
+  /**
+   * Converts an xpub to a compressed pub
+   * @param xpub
+   */
+  xpubToCompressedPub(xpub: string): string {
+    if (!this.isValidXpub(xpub)) {
+      throw new Error('invalid xpub');
+    }
+
+    const hdNode = utxoLib.HDNode.fromBase58(xpub, this.bitcoinEncoding());
+    return hdNode.keyPair.__Q.getEncoded(false).toString('hex');
+  }
+
+  compressedPubToHexAddress(pub: string): string {
+    const byteArrayAddr = bitgoAccountLib.Trx.Utils.getByteArrayFromHexAddress(pub);
+    const rawAddress = bitgoAccountLib.Trx.Utils.getRawAddressFromPubKey(byteArrayAddr);
+    return Buffer.from(rawAddress).toString('hex').toUpperCase();
+  }
+
+  xprvToCompressedPrv(xprv: string): string {
+    if (!this.isValidXprv(xprv)) {
+      throw new Error('invalid xprv');
+    }
+
+    const hdNode = utxoLib.HDNode.fromBase58(xprv, this.bitcoinEncoding());
+    return hdNode.keyPair.d.toBuffer(32).toString('hex');
+  };
+
+  bitcoinEncoding(): any {
+    return {
+      messagePrefix: '\x18Bitcoin Signed Message:\n',
+      bech32: 'bc',
+      bip32: {
+        public: 0x0488b21e,
+        private: 0x0488ade4
+      },
+      pubKeyHash: 0x00,
+      scriptHash: 0x05,
+      wif: 0x80,
+      coin: 'btc',
+    };
+  }
+
+  /**
+   * Make a query to Trongrid for information such as balance, token balance, solidity calls
+   * @param query {Object} key-value pairs of parameters to append after /api
+   * @param callback
+   * @returns {Object} response from Trongrid
+   */
+  recoveryPost(query: { path: string, jsonObj: any, node: NodeTypes }, callback?: NodeCallback<any>): Bluebird<any> {
```

Comment:
Can a more specific return type be used here instead of any?

Reply:
it varies based on the route unfortunately